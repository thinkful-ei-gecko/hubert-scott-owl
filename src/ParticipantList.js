import React from 'react';
import Participant from './Participant';

//props = array of participant objects
export default function ParticipantList(props) {
    let participants = props.participants.map(person => <Participant key = {person.id} avatar = {person.avatar} name = {person.name} inSession = {person.inSession} onStage = {person.onStage} />)

    return (
        <ul className="participant-list">
            {participants}
        </ul>
    )
}