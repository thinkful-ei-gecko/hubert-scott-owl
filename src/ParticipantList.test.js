import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantList from './ParticipantList'
import renderer from 'react-test-renderer';


describe('<ParticipantList />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<ParticipantList participants = {[    {
            id: 4,
            name: 'Frank Runciman',
            avatar:
                'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: false
        },
        {
            id: 5,
            name: 'Ashla Attwool',
            avatar:
                'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        }
    ]} />, div);

        ReactDOM.unmountComponentAtNode(div);
    });

});

describe('<ParticipantList />', () => {
    it('renders UI correctly', () => {
        const tree = renderer.create(<ParticipantList participants = {[
            {
            id: 4,
            name: 'Frank Runciman',
            avatar:
                'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: false
        },
        {
            id: 5,
            name: 'Ashla Attwool',
            avatar:
                'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        }
    ]} />).toJSON();
    
    expect(tree).toMatchSnapshot();
    });
});