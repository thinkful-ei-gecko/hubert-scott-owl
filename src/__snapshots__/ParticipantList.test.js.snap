// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantList /> renders UI correctly 1`] = `
<ul
  className="participant-list"
>
  <li
    className="participant"
  >
    <img
      alt="placeholder"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <h3>
      Frank Runciman
    </h3>
    <div
      className="in-session"
    />
    <p>
      
    </p>
  </li>
  <li
    className="participant"
  >
    <img
      alt="placeholder"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <h3>
      Ashla Attwool
    </h3>
    <div
      className="in-session"
    />
    <p>
      on stage
    </p>
  </li>
</ul>
`;
